File: problem_1/Python/p1.py
#!/usr/bin/env python

def counter(low, high):
    current = low
    while current < high:
        yield current
        current += 1

def iterator(begin, end):        
    num = 0
    for c in counter(begin, end):
        if c % 3 == 0:
            num += c
        elif c % 5 == 0:
            num += c
    return num

print  "answer => %d" % iterator(1, 1000)

File: problem_1/Java/P1.java
class P1 {
    public static void main(String[] args){
        int num = 0;
        for (int i = 1; i < 1000; i++ ) {
            if (i % 3 == 0) {
                num += i;
            } else if (i % 5 == 0) {
                num += i;
            }
        }
        System.out.printf("answer => %d\n",num);
    }
}
